tests/integration/snowflake/test_snowflake_adapter.py

```diff
@@ -0,0 +1,349 @@
+from tests.assets.integration_test_setup import CREDENTIALS, get_connection_profile
+from tests.common import query_equalize, rand_string
+
+
+@pytest.fixture(scope='session')
+def sf_adapter():
+    with open(CREDENTIALS) as cred_file:
+        credentials = yaml.safe_load(cred_file)
+
+    profile_dict = get_connection_profile(credentials)
+    profile_dict.update({"role": "snowshu_development_role"})
+    adapter = SnowflakeAdapter()
+    adapter.credentials = Credentials(**profile_dict)
+    return adapter
+
+
+def test_directionally_wrap_statement_directional(sf_adapter):
+    sampling = BernoulliSampleMethod(50, units='probability')
+    query = "SELECT * FROM highly_conditional_query"
+    DATABASE, SCHEMA, TABLE = [rand_string(10) for _ in range(3)]
+    relation = Relation(database=DATABASE,
+                        schema=SCHEMA,
+                        name=TABLE,
+                        materialization=TABLE,
+                        attributes=[])
+
+    assert query_equalize(sf_adapter.directionally_wrap_statement(query, relation, sampling)) == query_equalize(f"""
+            WITH
+                {relation.scoped_cte('SNOWSHU_FINAL_SAMPLE')} AS (
+            {query}
+            )
+            ,{relation.scoped_cte('SNOWSHU_DIRECTIONAL_SAMPLE')} AS (
+            SELECT
+                *
+            FROM
+                {relation.scoped_cte('SNOWSHU_FINAL_SAMPLE')}
+            SAMPLE BERNOULLI (50)
+            )
+            SELECT 
+                *
+            FROM 
+                {relation.scoped_cte('SNOWSHU_DIRECTIONAL_SAMPLE')}
+            """)
+
+
+def test_upstream_constraint_statement(sf_adapter):
+    DATABASE, SCHEMA, TABLE, LOCAL_KEY, REMOTE_KEY = [rand_string(10) for _ in range(5)]
+    relation = Relation(database=DATABASE,
+                        schema=SCHEMA,
+                        name=TABLE,
+                        materialization=TABLE,
+                        attributes=[])
+    statement = sf_adapter.upstream_constraint_statement(relation, LOCAL_KEY, REMOTE_KEY)
+
+    assert query_equalize(statement) == query_equalize(f" {LOCAL_KEY} in (SELECT {REMOTE_KEY} FROM \
+                {sf_adapter.quoted_dot_notation(relation)})")
+
+
+def test_population_count_statement(sf_adapter):
+    DATABASE, SCHEMA, TABLE = [rand_string(10) for _ in range(3)]
+    relation = Relation(database=DATABASE,
+                        schema=SCHEMA,
+                        name=TABLE,
+                        materialization=TABLE,
+                        attributes=[])
+    statement = sf_adapter.population_count_statement(relation)
+
+    assert query_equalize(statement) == query_equalize(
+        f"SELECT COUNT(*) FROM {sf_adapter.quoted_dot_notation(relation)}")
+
+
+def test_get_all_databases(sf_adapter):
+    databases_list = ["SNOWSHU_DEVELOPMENT", "HU_DATA", "SNOWFLAKE_SAMPLE_DATA", "DEMO_DB", "UTIL_DB"]
+
+    assert databases_list.sort() == sf_adapter._get_all_databases().sort()
```

Comment:
For simplicity we can probably limit this to only `SNOWSHU_DEVELOPMENT` for now since the `snowshu_development_role` should only have access to that database, or do we need to adjust the permissions for it?